packages/shared/src/logger.ts

```diff
@@ -106,6 +108,28 @@ class Logger {
     }
   }
 
+  private formatTags(tags?: Record<string, unknown>) {
+    if (!tags) {
+      return;
+    }
+
+    let result: Record<string, unknown> = {};
+
+    for (const [key, value] of Object.entries(tags)) {
+      if (value instanceof Error) {
+        result = {
+          ...result,
+          [key]: { errorName: value.name, errorMessage: value.message, errorStack: value.stack },
+        };
+        continue;
+      }
+
+      result = { ...result, [key]: value };
+    }
+
+    return result;
+  }
```

Comment:
nit: I avoid destructuring here. Log is used frequently, this adds overhead but probably not that much that's why nit.

Though I'd prefer to spread ...value, if in case there are more properties attached. Like if we extend an Error object and more properties in that.
```suggestion
  private formatTags(tags?: Record<string, unknown>) {
    if (!tags) {
      return;
    }
  
    return Object.entries(tags).reduce((result, [key, value]) => {
      if (value instanceof Error) {
        result[key] = {
          ...(value as Record<string, unknown>), // Intentionally keeping this for any extra properties attached in Error
          errorName: value.name,
          errorMessage: value.message,
          errorStack: value.stack,
        };
      } else {
        result[key] = value;
      }
      return result;
    }, {} as Record<string, unknown>);
  }
```